frame/assets-registry/src/lib.rs

```diff
@@ -190,298 +117,110 @@ pub mod pallet {
+		pub fn update_asset(
 			origin: OriginFor<T>,
-			foreign_admin: T::AccountId,
+			asset_id: T::LocalAssetId,
+			location: T::ForeignAssetId,
+			ratio: Option<Ratio>,
+			decimals: Option<Exponent>,
 		) -> DispatchResultWithPostInfo {
 			T::UpdateAdminOrigin::ensure_origin(origin)?;
-			<ForeignAdmin<T>>::put(foreign_admin.clone());
-			Self::deposit_event(Event::ForeignAdminUpdated(foreign_admin));
-			Ok(().into())
-		}
-
-		/// Approve an assets mapping candidate.
-		///
-		/// The dispatch origin for this call is `Config::LocalAdminOrigin` or
-		/// `Config::ForeignAdminOrigin`.
-		///
-		/// # Emits
-		///  - [`Event::AssetsMappingCandidateUpdated`](Event::AssetsMappingCandidateUpdated)
-		///
-		/// # Errors
-		/// - `OnlyAllowedForAdmins` when the origin isn't `Config::LocalAdminOrigin` or
-		///   `Config::ForeignAdminOrigin`.
-		/// - `LocalAssetIdAlreadyUsed` when local_asset_id is already used.
-		/// - `ForeignAssetIdAlreadyUsed` when foreign_asset_id is already used.
-		#[pallet::weight(<T as Config>::WeightInfo::approve_assets_mapping_candidate())]
-		pub fn approve_assets_mapping_candidate(
-			origin: OriginFor<T>,
-			local_asset_id: T::LocalAssetId,
-			foreign_asset_id: T::ForeignAssetId,
-			location: T::Location,
-			decimals: u8,
-		) -> DispatchResultWithPostInfo {
-			let who = ensure_signed(origin.clone())?;
-			Self::ensure_admins_only(origin)?;
-			ensure!(
-				!<LocalToForeign<T>>::contains_key(local_asset_id),
-				Error::<T>::LocalAssetIdAlreadyUsed
-			);
-			ensure!(
-				!<ForeignToLocal<T>>::contains_key(foreign_asset_id),
-				Error::<T>::ForeignAssetIdAlreadyUsed
-			);
-			Self::approve_candidate(who, local_asset_id, foreign_asset_id, location, decimals)?;
-			Self::deposit_event(Event::AssetsMappingCandidateUpdated {
-				local_asset_id,
-				foreign_asset_id,
-			});
-			Ok(().into())
-		}
-
-		/// Update a foreign metadata for an assets mapping.
-		///
-		/// The dispatch origin for this call is `Config::LocalAdminOrigin` or
-		/// `Config::ForeignAdminOrigin`.
-		///
-		/// # Emits
-		///  - [`Event::AssetMetadataUpdated`](Event::AssetMetadataUpdated)
-		///
-		/// # Errors
-		/// - `OnlyAllowedForAdmins` when the origin isn't `Config::LocalAdminOrigin` or
-		///   `Config::ForeignAdminOrigin`.
-		/// - `LocalAssetIdNotFound` when local_asset_id not found.
-		#[pallet::weight(<T as Config>::WeightInfo::set_metadata())]
-		pub fn set_metadata(
-			origin: OriginFor<T>,
-			local_asset_id: T::LocalAssetId,
-			metadata: ForeignMetadata,
-		) -> DispatchResultWithPostInfo {
-			let _ = ensure_signed(origin.clone())?;
-			Self::ensure_admins_only(origin)?;
-			ensure!(
-				<LocalToForeign<T>>::contains_key(local_asset_id),
-				Error::<T>::LocalAssetIdNotFound
-			);
-
-			<ForeignAssetMetadata<T>>::insert(local_asset_id, metadata);
-			Self::deposit_event(Event::AssetMetadataUpdated(local_asset_id));
+			// note: does not validates if assets exists, not clear what is expected in this case
```

Comment:
can add
```
ensure!(
  <LocalToForeign<T>>::contains_key(asset_id),
  Error::<T>::LocalAssetIdNotFound
);
```

Reply:
second ensure, really it may be ok to have different ids for same remote.
first, we add assets which are special id from well known, but well known assets are no need to be in list. will attach link to clickup to solve well knowns as usual. than can do check number 1. also as adding is pemissioned, its not a security  to add like that.

Reply:
as of not method is ok, can fix first clause after https://app.clickup.com/t/1wtyb38 . second clause is ok as used to handle well knowns.